### PR TITLE
Fix for `applyHook()` to pass `$hookArg` to all registered callbacks

### PR DESCRIPTION
**Pull request by ghost**
I addressed an issue in applyHook() where only the first callback was receiving the specified $hookArg parameter. 

Also, applyHook() no longer returns anything.

### CODE REVIEW

File: Slim/Slim.php [hunk 1]
```diff
@@ -1002,7 +1002,7 @@ public function hook( $name, $callable, $priority = 10 ) {
      * Invoke hook
      * @param   string  $name       The hook name
      * @param   mixed   $hookArgs   (Optional) Argument for hooked functions
-     * @return  mixed
+     * @return  void
```
Comment (ghost) [inline Slim/Slim.php:1005]: Removed `return $hookArg;` so now the method no longer returns anything.

File: Slim/Slim.php [hunk 2]
```diff
@@ -1016,11 +1016,12 @@ public function applyHook( $name, $hookArg = null ) {
             foreach( $this->hooks[$name] as $priority ) {
                 if( !empty($priority) ) {
                     foreach($priority as $callable) {
-                        $hookArg = call_user_func($callable, $hookArg);
+                        call_user_func($callable, $hookArg);
                     }
                 }
             }
-            return $hookArg; } }
```
Comment (ghost) [inline Slim/Slim.php:1023]: Removed this line per our discussion:

`There should be no return value for`applyHook()`. If someone really wants each registered callback for a given hook to update the argument, that can still be  accomplished by passing an object reference.`

File: Slim/Slim.php [hunk 3]
```diff
@@ -1016,11 +1016,12 @@ public function applyHook( $name, $hookArg = null ) {
             foreach( $this->hooks[$name] as $priority ) {
                 if( !empty($priority) ) {
                     foreach($priority as $callable) {
-                        $hookArg = call_user_func($callable, $hookArg);
+                        call_user_func($callable, $hookArg);
```
Comment (ghost) [inline Slim/Slim.php:1019]: Removed the `$hookArg` assignment and simply call the function.

**Comment by codeguy** [comment]
Thanks! Cherry-picked this into the release-1.6.3 branch. Will roll into master sometime this week.

**Comment by ghost** [comment]
Awesome, glad I could help!

**Pull request closed without merging**